nifi-nar-bundles/nifi-morphlines-bundle/nifi-morphlines-processor/src/main/java/org/apache/nifi/processors/morphlines/Morphlines.java

```diff
@@ -0,0 +1,214 @@
+            .required(false)
+            .defaultValue("_attachment_body")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Relationship for success.")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Relationship for failure of morphlines.")
+            .build();
+
+    public static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("Relationship for original flowfiles.")
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = ImmutableList.<PropertyDescriptor>builder()
+            .add(MORPHLINES_FILE)
+            .add(MORPHLINES_ID)
+            .add(MORPHLINES_OUTPUT_FIELD)
+            .build();
+
+    private static final Set<Relationship> RELATIONSHIPS = ImmutableSet.<Relationship>builder()
+            .add(REL_SUCCESS)
+            .add(REL_FAILURE)
+            .build();
+
+    private File morphLineFile;
+    private String morphLinesId;
+    private String morphlinesOutputField;
+    private MorphlineContext morphlineContext;
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) throws Exception {
+        morphLineFile = new File(context.getProperty(MORPHLINES_FILE).getValue());
+        morphLinesId = context.getProperty(MORPHLINES_ID).getValue();
+        morphlinesOutputField = context.getProperty(MORPHLINES_OUTPUT_FIELD).getValue();
+        morphlineContext = new MorphlineContext.Builder().build();
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        final AtomicLong written = new AtomicLong(0L);
+        final byte[] value = new byte[(int) flowFile.getSize()];
+
+        try{
+            flowFile = session.write(flowFile, new StreamCallback() {
+                @Override
+                public void process(InputStream in, OutputStream out) throws IOException {
+                    StreamUtils.fillBuffer(in, value);
+                    Record record = new Record();
+                    record.put(Fields.ATTACHMENT_BODY, value);
+                    Collector collectorRecord = new Collector();
+                    morphline = new org.kitesdk.morphline.base.Compiler().compile(morphLineFile, morphLinesId, morphlineContext, collectorRecord);
```

Comment:
(Without knowing much about Morphlines) This looks very inefficient to compile the Morphline for every incoming FlowFile. Is this how it's supposed to be done or can we move the compiling to the OnScheduled method?